test/libsolidity/semanticTests/using/module_1.sol

```diff
@@ -0,0 +1,18 @@
+==== Source: A ====
+function id(uint x) pure returns (uint) {
+    return x;
+}
+
+==== Source: B ====
+import "A" as M;
+contract C {
+    using M for uint;
```

Comment:
If the import statement above looks like `import "A.sol";` (`A.sol` being the file instead of simply `A`) and then we do `using A.sol for uint`, does the parser parse it as intended?

In other words, if no module name is defined, does it default the module name default to file name?

Reply:
No, it does not work like that at all. `import "A.sol";` imports all top-level elements from `"A.sol"` into the current scope, while `import "A.sol" as M;` creates a new top-level scope called `M` that you can reference.

Reply:
In the second syntax you need to access items as `M.f` while in the first you just use `f`.